src/backend/utils/adt/arrayfuncs.c

```diff
@@ -3060,7 +3060,7 @@ construct_md_array(Datum *elems,
 	memcpy(ARR_DIMS(result), dims, ndims * sizeof(int));
 	memcpy(ARR_LBOUND(result), lbs, ndims * sizeof(int));
 
-	if (elems==NULL && fixedwidthtype) 
+	if (elems==NULL || fixedwidthtype) 
```

Comment:
I suggest a change. It seems weird that we don't copy elements if they are of fixed width.
```suggestion
	if (elems==NULL)
```
There is some info about this in the comment atop of this function but I found it a little confusing. Also, `fixedwidthtype` is Greenplum-specific code.

Reply:
> I suggest a change. It seems weird that we don't copy elements if they are of fixed width.
> 
> There is some info about this in the comment atop of this function but I found it a little confusing. Also, `fixedwidthtype` is Greenplum-specific code.

Also working idea